core/src/main/java/org/kiwix/kiwixmobile/core/main/CoreReaderFragment.java

```diff
@@ -834,18 +833,20 @@ protected void restoreDeletedTab(int index) {
 
   protected void selectTab(int position) {
     currentWebViewIndex = position;
-    contentFrame.removeAllViews();
-    KiwixWebView webView = safelyGetWebView(position);
-    if (webView.getParent() != null) {
-      ((ViewGroup) webView.getParent()).removeView(webView);
-    }
-    contentFrame.addView(webView);
-    tabsAdapter.setSelected(currentWebViewIndex);
-    updateBottomToolbarVisibility();
-    loadPrefs();
-    updateUrlProcessor();
-    updateTableOfContents();
-    updateTitle();
+    //if (contentFrame != null)
+      contentFrame.removeAllViews();
+      KiwixWebView webView = safelyGetWebView(position);
+      if (webView.getParent() != null) {
+        ((ViewGroup) webView.getParent()).removeView(webView);
+      }
+      contentFrame.addView(webView);
+      tabsAdapter.setSelected(currentWebViewIndex);
+      updateBottomToolbarVisibility();
+      loadPrefs();
+      updateUrlProcessor();
+      updateTableOfContents();
+      updateTitle();
+    //}
```

Comment:
When this null check is enabled, it freezes the back functionality on the download screen but in now(when disabled) crashes the app due to npe at `contentFrame`